packages/flutter/lib/src/widgets/dismissable.dart

```diff
@@ -114,6 +114,43 @@ class Dismissable extends StatefulWidget {
   _DismissableState createState() => new _DismissableState();
 }
 
+class _DismissableClipper extends CustomClipper<Rect> {
+  _DismissableClipper({
+    this.axis,
+    Animation<FractionalOffset> moveAnimation
+  }) : moveAnimation = moveAnimation, super(reclip: moveAnimation) {
+    assert(axis != null);
+    assert(moveAnimation != null);
+  }
+
+  final Axis axis;
+  final Animation<FractionalOffset> moveAnimation;
+
+  Rect getClip(Size size) {
+    assert(axis != null);
+    switch (axis) {
+      case Axis.horizontal:
+        final double offset = moveAnimation.value.dx * size.width;
+        if (offset < 0)
+          return new Rect.fromLTRB(size.width + offset, 0.0, size.width, size.height);
+        return new Rect.fromLTRB(0.0, 0.0, offset, size.height);
+      case Axis.vertical:
+        final double offset = moveAnimation.value.dy * size.height;
+        if (offset < 0)
+          return new Rect.fromLTRB(0.0, size.height + offset, size.width, size.height);
+        return new Rect.fromLTRB(0.0, 0.0, size.width, offset);
+    }
+    return null;
+  }
+
+  Rect getApproximateClipRect(Size size) => getClip(size);
+
+  bool shouldReclip(_DismissableClipper oldClipper) {
+    return oldClipper.axis != axis
+        || oldClipper.moveAnimation != moveAnimation;
```

Comment:
Good idea.  Doesn't actually matter in this specific case, but that's a better practice that we should model.